spark/src/test/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuerySuite.scala

```diff
@@ -188,99 +644,249 @@ class OptimizeMetadataOnlyDeltaQuerySuite
+    assert(showResultData.size === 1)
+    assert(showResultData.head.head.getString(0).toLong === totalRows)
+
+    val showMultAggPlans = DeltaTestUtils.withLogicalPlansCaptured(spark, optimizedPlan = true) {
+      spark.sql(s"SELECT COUNT(*), MIN(id), MAX(id) FROM $testTableName").show()
+    }
+
+    val showMultipleAggResultData = showMultAggPlans.collect { case x: LocalRelation => x.data }
+    assert(showMultipleAggResultData.size === 1)
+    val firstRow = showMultipleAggResultData.head.head
+    assert(firstRow.getString(0).toLong === totalRows)
+    assert(firstRow.getString(1).toLong === minId)
+    assert(firstRow.getString(2).toLong === maxId)
   }
 
-  test("Select Count: group by") {
-    checkSameQueryPlanAndResults(
-      s"SELECT group, COUNT(*) FROM $testTableName GROUP BY group",
-      Seq(Row(1L, totalRows)))
+  test("min-max .show() - only NULL values") {
+    val tableName = "TestOnlyNullValuesShow"
+
+    spark.sql(s"CREATE TABLE $tableName (Column1 INT) USING DELTA")
+    spark.sql(s"INSERT INTO $tableName (Column1) VALUES (NULL);")
+
+    val showMultAggPlans = DeltaTestUtils.withLogicalPlansCaptured(spark, optimizedPlan = true) {
+      spark.sql(s"SELECT MIN(Column1), MAX(Column1) FROM $tableName").show()
+    }
+
+    val showMultipleAggResultData = showMultAggPlans.collect { case x: LocalRelation => x.data }
+    assert(showMultipleAggResultData.size === 1)
+    val firstRow = showMultipleAggResultData.head.head
+    assert(firstRow.getString(0) === "NULL")
+    assert(firstRow.getString(1) === "NULL")
   }
 
-  test("Select Count: count twice") {
-    checkSameQueryPlanAndResults(
-      s"SELECT COUNT(*), COUNT(*) FROM $testTableName",
-      Seq(Row(totalRows, totalRows)))
+  test("min-max .show() - Date Columns") {
+    val tableName = "TestDateColumnsShow"
+
+    spark.sql(s"CREATE TABLE $tableName (Column1 DATE, Column2 DATE) USING DELTA")
+    spark.sql(s"INSERT INTO $tableName (Column1, Column2) VALUES " +
+      s"(CAST('1582-10-15' AS DATE), NULL);")
+
+    val showMultAggPlans = DeltaTestUtils.withLogicalPlansCaptured(spark, optimizedPlan = true) {
+      spark.sql(s"SELECT MIN(Column1), MIN(Column2) FROM $tableName").show()
+    }
+
+    val showMultipleAggResultData = showMultAggPlans.collect { case x: LocalRelation => x.data }
+    assert(showMultipleAggResultData.size === 1)
+    val firstRow = showMultipleAggResultData.head.head
+    assert(firstRow.getString(0) === "1582-10-15")
+    assert(firstRow.getString(1) === "NULL")
   }
 
-  test("Select Count: plus literal") {
-    checkSameQueryPlanAndResults(
-      s"SELECT COUNT(*) + 1 FROM $testTableName",
-      Seq(Row(totalRows + 1)))
+  // Tests to validate the optimizer won't incorrectly change queries it can't correctly handle
+
+  Seq((s"SELECT COUNT(*) FROM $mixedStatsTableName", "missing stats"),
+    (s"SELECT COUNT(*) FROM $noStatsTableName", "missing stats"),
```

Comment:
few tests to add (apologies if they are already present)
1) table with DVs
2) empty table
3) table with few AddFiles having zero rows

Reply:
Done.
We have an existing test for count + empty table. I added the other cases.
For min/max + empty table, that case is not supported. (theoretically it could be supported, but I preferred to be safe and don't optimize, it won't read any parquet anyway)